### DISCUSSION TITLE
Use theme with Jekyll 4.x

### DISCUSSION THREAD

**Discussion by big-andy-coates** [mmistakes/minimal-mistakes#4007] [Q&A] [answered]
Hi everyone,

Is is possible to use this theme with Jekyll 4.x?  On Github pages site?

Love this theme and would like to use it as the basis for an open-source software site, which will be built using GitHub pages across several different GitHub repositories.  To keep the site consistent, I'd like to fork this repo and customise with some defaults, e.g. title in `_condfig.yml`, masthead nav in `_data/navigation/main.yml`, etc.  

Unfortunately, using Jekyll 3.9.2, which the `github-pages` gem requires, `_config.yml` and `_data` aren't loaded from remote themes.  Hence, I'd like to upgrade to Jekyll 4.x, which does load these resources.

Does anyone know:
* is this theme compatible with Jekyll 4.x?  (I see there are some [breaking changes](https://jekyllrb.com/docs/upgrading/3-to-4/))
* what I need to change in my fork to get this working with Jekyll 4.x locally and on GitHub pages?  I'm assuming I need to drop the `github-pages` gem and switch to a GitHub workflow to build and deploy the site... 

I'm a ruby / Jekyll noob... and my Googling isn't throwing up much in the way of steps and examples, and my own attempts have quickly resulted in build errors I don't know how to fix.

Help and pointers welcome!

**Comment by LindseyB** [comment, marked as answer]
You can deploy github pages using the github pages action, by going to your repo with the project you want to deploy. Going to settings and changing the source to GitHub actions.

![chrome_uqE3nL5ELd](https://user-images.githubusercontent.com/33750/201549105-4c03e746-1c81-4bee-ad3a-16d028a3f45b.png) 

The action really shouldn't require any changes from you.

From there you can deploy using the [gem based method](https://mmistakes.github.io/minimal-mistakes/docs/quick-start-guide/#gem-based-method) and it works great. 

I actually have an example over at https://github.com/LindseyB/blog-jekyll doing exactly this.
